### PR TITLE
add ChainUserProvider

### LINKED ISSUE #615: Chain providers
A bug when adding more than one providers to the firewalls
[local_history.patch.txt](https://github.com/lexik/LexikJWTAuthenticationBundle/files/2841534/local_history.patch.txt)

### LINKED ISSUE #754: loadUserByUsernameAndPayload is never called in my Custom User Providers in chain configuration
Hi,

I have two distinct Custom User Providers for two different user types ClientUserProvider and CustomerUserProvider, each of this provider implements PayloadAwareUserProviderInterface when I used either of the providers in the firewall the method is called but when I configured the two to be a chain and use it in the firewall the loadUserByUsernameAndPayload is never called it only calls the default loadUserByUsername.

```
providers:
	client_security_user_provider:
	  id: App\Security\ClientUserProvider
	customer_security_user_provider:
	  id: App\Security\CustomerUserProvider
	chained_security_user_provider:
		chain:
			providers: ['client_security_user_provider','customer_security_user_provider']
			
	# customer firewall    
	api_customer:
		pattern:    ^/v1/customer
		stateless: true
		anonymous: false
		provider:  customer_security_user_provider
		guard:
			authenticators:
				- lexik_jwt_authentication.jwt_token_authenticator
				
	# client firewall    
	api_customer:
		pattern:    ^/v1/client
		stateless: true
		anonymous: false
		provider:  client_security_user_provider
		guard:
			authenticators:
				- lexik_jwt_authentication.jwt_token_authenticator
				
	# client and customer shared firewall    
	api_customer:
		pattern:    ^/v1/common
		stateless: true
		anonymous: false
		provider:  chained_security_user_provider
		guard:
			authenticators:
				- lexik_jwt_authentication.jwt_token_authenticator
```

### PR DESCRIPTION
**Pull request by Gemorroj**
Fixes #615
Fixes #754

### CODE REVIEW

**Comment by chalasr** [comment]
Thanks for fixing this bug @Gemorroj.

File: Security/Guard/JWTTokenAuthenticator.php [hunk 1]
```diff
@@ -285,6 +286,14 @@ protected function loadUser(UserProviderInterface $userProvider, array $payload,
             return $userProvider->loadUserByUsernameAndPayload($identity, $payload);
         }
 
+        if ($userProvider instanceof ChainUserProvider) {
+            foreach ($userProvider->getProviders() as $provider) {
+                if ($provider instanceof PayloadAwareUserProviderInterface) {
```
Comment (CvekCoding) [inline Security/Guard/JWTTokenAuthenticator.php:291]: It's better to wrap this part into `try-catch` like it's done in original `Chain` provider https://github.com/symfony/security/blob/9b4060f040be91d55db67831bf83981097c609b1/Core/User/ChainUserProvider.php#L52

Comment (chalasr) [reply Security/Guard/JWTTokenAuthenticator.php:291]: @CvekCoding good catch. Would you mind sending a PR or shall I do it?

Comment (CvekCoding) [reply Security/Guard/JWTTokenAuthenticator.php:291]: Sure, I'll do this today.

Comment (CvekCoding) [reply Security/Guard/JWTTokenAuthenticator.php:291]: @chalasr when this fix can be released?

Comment (chalasr) [reply Security/Guard/JWTTokenAuthenticator.php:291]: @CvekCoding it has been released in https://github.com/lexik/LexikJWTAuthenticationBundle/releases/tag/v2.8.0

Comment (CvekCoding) [reply Security/Guard/JWTTokenAuthenticator.php:291]: Done https://github.com/lexik/LexikJWTAuthenticationBundle/pull/766

**Pull request merged**